crates/orchestrator/src/config.rs

```diff
@@ -159,25 +159,140 @@ async fn build_da_client() -> Box<dyn DaClient + Send + Sync> {
 }
 
 /// Builds the prover service based on the environment variable PROVER_SERVICE
-fn build_prover_service(settings_provider: &impl SettingsProvider) -> Box<dyn ProverClient> {
+pub fn build_prover_service(settings_provider: &impl SettingsProvider) -> Box<dyn ProverClient> {
     match get_env_var_or_panic("PROVER_SERVICE").as_str() {
         "sharp" => Box::new(SharpProverService::with_settings(settings_provider)),
         _ => panic!("Unsupported prover service"),
     }
 }
 
 /// Builds the settlement client depending on the env variable SETTLEMENT_LAYER
-async fn build_settlement_client(settings_provider: &impl SettingsProvider) -> Box<dyn SettlementClient + Send + Sync> {
+pub async fn build_settlement_client(
+    settings_provider: &impl SettingsProvider,
+) -> Box<dyn SettlementClient + Send + Sync> {
     match get_env_var_or_panic("SETTLEMENT_LAYER").as_str() {
         "ethereum" => Box::new(EthereumSettlementClient::with_settings(settings_provider)),
         "starknet" => Box::new(StarknetSettlementClient::with_settings(settings_provider).await),
         _ => panic!("Unsupported Settlement layer"),
     }
 }
 
-async fn build_storage_client() -> Box<dyn DataStorage + Send + Sync> {
+pub async fn build_storage_client() -> Box<dyn DataStorage + Send + Sync> {
     match get_env_var_or_panic("DATA_STORAGE").as_str() {
         "s3" => Box::new(AWSS3::new(AWSS3Config::new_from_env()).await),
         _ => panic!("Unsupported Storage Client"),
     }
 }
+
+#[cfg(test)]
```

Comment:
Nitpick: why don't you put this in a dedicated module to avoid annotating every item with `#[cfg(test)]` ?

Reply:
Valid, Moved this code to a `orchestrator/src/tests/config.rs` 
No need to feature flag now.